sql/expression/function/sleep.go

```diff
@@ -37,8 +37,12 @@ func (s *Sleep) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 		return nil, err
 	}
 
-	time.Sleep(time.Duration(child.(float64)*1000) * time.Millisecond)
-	return 0, nil
+	select {
+	case <-time.After(time.Duration(child.(float64) * 1000) * time.Millisecond):
+		return 0, nil
+	case <-ctx.Done():
+		return 0, nil
```

Comment:
I would return `context.Canceled` here as the error

Reply:
Done.